libs/vm/include/vm/variant.hpp

```diff
@@ -532,4 +534,210 @@ struct AnyFloatingPoint : Variant
 };
 
 }  // namespace vm
+
+namespace serializers {
+
+template <typename D>
+struct MapSerializer<fetch::vm::Variant, D>
+{
+public:
+  using Type       = fetch::vm::Variant;
+  using DriverType = D;
+
+  static uint8_t const TYPEID    = 1;
+  static uint8_t const PRIMITIVE = 2;
+  static uint8_t const OBJECT    = 3;
+
+  template <typename Constructor>
+  static void Serialize(Constructor &map_constructor, Type const &variant)
+  {
+    auto map = map_constructor(3);
+    map.Append(TYPEID, variant.type_id);
+
+    // empty variant
+    if (variant.type_id == vm::TypeIds::Unknown)
+    {
+      return;
+    }
+
+    // primitive type variant
+    if (variant.IsPrimitive())
+    {
+      switch (variant.type_id)
+      {
+      case vm::TypeIds::Bool:
+      case vm::TypeIds::UInt8:
```

Comment:
Since underlying message pack serialisation only really cares about unsigned and signed types. Might make sense to reflect this in this code and case all unsigned values to `uint16_t` and all signed values to `int64_t`

Reply:
I am not sure doing so will simplify the code, as we will have to add `static_cast`s for every type in the `Primitive` field